### PR TITLE
Add initial sentry support

### PR DESCRIPTION
**Pull request by hsiW**
This PR is not complete yet, once I know that the stuff I've added/done so far is good I'll continue to update it.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -15,6 +15,23 @@ version = "2.26.2"
...
+version = "0.1"
+
+[features]
+default = [
+  "sentry_logger"
+]
+sentry_logger = [
+  "sentry",
+  "tokio-core"
+]
```
Comment (zeyla) [inline Cargo.toml]: eof

File: Cargo.toml [hunk 2]
```diff
@@ -15,6 +15,23 @@ version = "2.26.2"
...
 git = "https://github.com/zeyla/serenity"
+
+[dependencies.tokio-core]
+optional = true
+version = "0.1"
+
+[features]
+default = [
+  "sentry_logger"
+]
```
Comment (zeyla) [inline Cargo.toml]: These arrays can be single line

File: cli.yml [hunk 3]
```diff
@@ -3,27 +3,33 @@ version: "0.1.0"
...
       help: Roles that are allowed to run commands
-  - message:
-      short: m
-      long: message
+  - sentry:
+      short: s
+      long: sentry
+      required: false
+      takes_value: true
+      help:
```
Comment (zeyla) [inline cli.yml]: `help` is empty

File: src/handler.rs [hunk 4]
```diff
@@ -5,7 +5,7 @@ use serenity::model::{Message, Ready};
 use super::store::*;
 
 lazy_static! {
-    static ref PATTERN: Regex = Regex::new("^(!{3,}|;+|\.{3,})[A-Za-z]+").unwrap();
+    static ref PATTERN: Regex = Regex::new("^(!{3,}|;+|.{3,})[A-Za-z]+").unwrap();
```
Comment (zeyla) [inline src/handler.rs]: This changes the behaviour of the regex string

File: src/main.rs [hunk 5]
```diff
@@ -76,6 +91,16 @@ fn main() {
...
 }
 
+#[cfg(feature = "sentry_logger")]
+fn sentry(dns_url: &str) -> Sentry {
+    let core = Core::new();
+    let handle = core.unwrap().handle();
+
+    let credential: SentryCredential = dns_url.parse().unwrap();
+
+    Sentry::from_settings(handle, Settings::default(), credential)
```
Comment (zeyla) [inline src/main.rs]: This won't work since the core is dropped and never ran.

**Comment by hsiW** [comment]
This PR will basically be paused until the serenity-rust lib isn't so terrible

**Pull request closed without merging**